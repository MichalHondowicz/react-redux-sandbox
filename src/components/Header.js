import React from 'react';
import {Link} from "react-router-dom";

class Header extends React.Component {

    constructor() {
        super();
        this.state = {
            bookstoreName: 'Magic books',
            clicked: true,
            textColor: 'white',
            backgroundColor: 'black'
        }
    }

    handleClicked = () => {

        if (this.state.clicked) {
            this.setState({
                bookstoreName: 'Black books',
                textColor: 'black',
                backgroundColor: 'white'
            })
        } else {
            this.setState({
                bookstoreName: 'Magic books',
                textColor: 'white',
                backgroundColor: 'black'
            })
        }
        this.setState({
            clicked: !this.state.clicked
        })
    };

    render() {
        let headerCss = {
            color: this.state.textColor,
            backgroundColor: this.state.backgroundColor,
        };
        return (
            <div style={headerCss} className="ui center aligned container" onClick={this.handleClicked}>
                <h1>{this.state.bookstoreName}</h1><br/>
                <Link to='/admin' className='ui right aligned container'>
                    <button className='ui button'>Admin Panel</button>
                </Link>
            </div>
        );
    }
}

export default Header;