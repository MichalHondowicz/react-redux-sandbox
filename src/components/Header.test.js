import ReactDOM from "react-dom";
import Header from "./Header";
import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'

configure({adapter: new Adapter()});

describe('Header test', () => {
    it('Header renders with no problems ', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Header/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Header changed text when setState', () => {
        const wrapper = shallow(<Header/>);
        expect(wrapper.find('h1').text()).toBe('Magic books');
        wrapper.setState({
            bookstoreName: 'Black books'
        });
        expect(wrapper.find('h1').text()).toBe('Black books');
    });

    it('Header state changed with onClick', () => {
        const wrapper = shallow(<Header/>);
        expect(wrapper.state().bookstoreName).toBe('Magic books');
        wrapper.find('.header').simulate('click');
        expect(wrapper.state().bookstoreName).toBe('Black books');
    });

    it('Snapshot ok', () => {
        const wrapper = shallow(<Header/>);
        expect(wrapper).toMatchSnapshot();
    })
});