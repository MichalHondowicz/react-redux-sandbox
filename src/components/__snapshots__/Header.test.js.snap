// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test Snapshot ok 1`] = `ShallowWrapper {}`;
